### DISCUSSION TITLE
Yarn 3 build cache is not working

### DISCUSSION THREAD

**Discussion by yikkwongwong** [vercel/community#1693] [Help]
### Summary

Is there anyway we can use .yarn/cache in Yarn 3? The folder is not being cached by Vercel.

Related:
https://github.com/orgs/vercel/discussions/222#discussion-3828997

### Additional information

.yarn directory after yarn install (first deployment)
![image](https://user-images.githubusercontent.com/25247355/222972070-1cbba638-1638-46ce-9944-653f2c4a572f.png)

.yarn directory before yarn install (second deployment), shows that cache folder is being removed
![image](https://user-images.githubusercontent.com/25247355/222972096-c0848a17-3f03-48a6-80d8-54b21f48f718.png)

**Comment by pepijn-vanvlaanderen** [comment]
Unfortunately this indeed does not work atm. It seems to be a quick resolution if Vercel adds caching for the default yarn cache folder, which is .yarn/cache.

As a workaround for NextJs projects you can alter the cache dir like this in the install command in Vercel:
```
YARN_CACHE_FOLDER=./.next/cache/yarn-cache yarn install
```
An example of the install command when you use Turborepo with a NextJs app called `website`:
```
cd ../.. && YARN_CACHE_FOLDER=./.next/cache/yarn-cache yarn workspace website install
```

For ViteJs projects I couldn't figure out a workaround so far, maybe someone else has something for that or Vercel will fix this soon.
